### DISCUSSION TITLE
SSO returns different PK than from a few hours ago

### DISCUSSION THREAD

**Discussion by ealeksandrov7** [Web3Auth/Web3Auth#614] [Help]
I experienced a weird thing. I've been trying to implement a self host solution and I've been playing around with a bunch of private keys, generating shares, etc. Until today - a few hours ago ALL of the SSOs I've been using for testing purposes suddenly started returning  different private keys in the login response. So I am not able to access the previously constructed keys.. 

E.G. I login with a new google account for the first time -> generate a share for the SecurityQuestions module -> generate another share which I store locally = I end up with 2/4 shares. I've been able to reconstruct these private keys by providing the shares up until now. Now all of my google accounts started returning a different PK in the login response, hence I am unable to reconstruct the same keys I've been previously using. If that was on a production app, these wallets would have been lost forever. So my question is:

Is there something **really** wrong with my implementation or is this because of some "testnet flush" or whatever?
Is it not **provided** that each SSO login would **always** provide the **same** private key?

**Comment by YZhenY** [comment]
Hard to tell without a code snippet of your implementation, provide that and we can dig deeper.

 With the same SSO you should get the same "Postbox Key" or "OAuth Key" every time, if you aren't, check the network your verifier is on is that the same? These are some reasons why your key might have reset/registered as new:
- does your implementation call "_initializeNewKey"? This resets your key, and should not be used in any case unless you know what youre doing.
- Did you change the storage layer? This makes it such the SDK think you're a new user.

**Reply by ealeksandrov7** [reply]
Thanks for your response, YZhenY!

So first I am triggering a login with:
![Screen Shot 2022-09-15 at 21 17 00](https://user-images.githubusercontent.com/112856139/190480477-ce7dbcbd-19e0-48f8-8915-b5048e41bd2e.png)

After that I am calling: 
![Screen Shot 2022-09-15 at 21 17 32](https://user-images.githubusercontent.com/112856139/190480594-dbeb193d-dc8c-4556-b7a9-e13dd17434f0.png)

Then I am setting a password with something like this:
![Screen Shot 2022-09-15 at 21 23 44](https://user-images.githubusercontent.com/112856139/190481070-2ac3c092-435f-4e98-a8fb-f67f0ffd1573.png)

And that's my modules initialization:
![Screen Shot 2022-09-15 at 21 25 16](https://user-images.githubusercontent.com/112856139/190481435-84766581-0162-480e-a45b-25b790fd0042.png)

Is there anything out of the ordinary? Am I right to use the private key coming from the SSO as a postBoxKey? My understanding was that this way I am using the private key from the SSO as the "1st share", but maybe I'm wrong.

**Reply by yashovardhan** [reply]
@1swaraj can you help us out here

**Reply by YZhenY** [reply]
nothing out of the ordinary here, 

so the post box key youre getting back is different? without any config change?

**Reply by ealeksandrov7** [reply]
I did 2 things actually:
  1. I tried adding/removing the WebStorage module, because I was trying to figure out how to to get the device share for my react native app. (btw no luck there, so I decided on generating a new share and storing it in a local secure storage - is that a bad idea?)
  2. I went from custom verifiers to the "built-in" ones
  
  Other than that I haven't changed the storage layer nor called the _initializeNewKey method.
  Hmm, if that doesn't make sense, I guess I'll continue testing it and come back to you if this happens again. Maybe I'll have more info 
  then

**Reply by YZhenY** [reply]
If you changed your verifier, you'll have changed the PK you get back as the postbox key

Verifiers are scoped to auth, and each verifier gives a different PK - that would be the reason!
